code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.test.ts

```diff
@@ -0,0 +1,189 @@
+
+  describe("with filters", () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    test("applies timespan filter correctly", () => {
+      const mockTimespan = { start: "2021-01-01", end: "2021-01-31" };
+      mockUseFiltersWithApply.mockReturnValueOnce({
+        filters: {},
+        appliedFilters: { timespan: mockTimespan },
+        activeFiltersCount: 1,
+        actions: {}
+      });
+      mockTimeSpanToDateRange.mockReturnValueOnce({
+        start: new Date(mockTimespan.start),
+        end: new Date(mockTimespan.end)
+      });
+
+      const { result } = renderHook(() => useFilteredAdminOrders());
+      expect(mockTimeSpanToDateRange).toHaveBeenCalledWith(mockTimespan);
+      expect(result.current).toBeDefined();
+    });
+
+    test("undefined timespan results in default state", () => {
+      const { result } = renderHook(useFilteredAdminOrders);
+      expect(result.current).toMatchObject({
+        activeFiltersCount: 0,
+        isLoading: false,
+        orders: []
+      });
+    });
+
+    test("applies delivery-methods filter correctly", () => {
+      const deliveryMethods = new Set(["method1", "method2"]);
+      const expectedOrders = [
+        { id: 1, name: "Order 1", deliveryMethod: "method1" }
+      ];
+
+      mockUseFiltersWithApply.mockReturnValueOnce({
+        filters: {},
+        appliedFilters: { "delivery-methods": deliveryMethods },
+        activeFiltersCount: 1,
+        actions: {}
+      });
+
+      mockUseGetAdminOrdersQuery.mockReturnValueOnce({
+        data: { content: expectedOrders },
+        isLoading: false
+      });
+
+      const { result } = renderHook(() => useFilteredAdminOrders());
+      expect(result.current.orders).toEqual(expectedOrders);
+    });
+
+    test("converts statuses set to array when provided", () => {
+      const statusesSet = new Set(["PENDING", "SHIPPED"]);
+      mockUseFiltersWithApply.mockReturnValueOnce({
+        filters: {},
+        appliedFilters: { statuses: statusesSet },
+        activeFiltersCount: 1,
+        actions: {}
+      });
+
+      mockUseGetAdminOrdersQuery.mockImplementation((queryParameters) => {
+        expect(queryParameters.statuses).toEqual(Array.from(statusesSet));
+        return { data: { content: [] }, isLoading: false };
+      });
+
+      renderHook(() => useFilteredAdminOrders());
+    });
+
+    test("handles empty delivery-methods set correctly", () => {
+      mockUseFiltersWithApply.mockReturnValueOnce({
+        filters: {},
+        appliedFilters: { "delivery-methods": new Set() },
+        activeFiltersCount: 0,
+        actions: {}
+      });
+
+      const { result } = renderHook(() => useFilteredAdminOrders());
+      expect(mockUseGetAdminOrdersQuery).toHaveBeenCalledWith({
+        isPaid: undefined,
+        totalLess: undefined,
+        totalMore: undefined,
+        statuses: undefined,
+        deliveryMethods: [],
+        createdBefore: undefined,
+        createdAfter: undefined
```

Comment:
This should be improved

Reply:
I moved it to a separate variable. Do you have another idea?

Reply:
I think you have some typescript error so you did it. I suggest defining only `deliveryMethods` and do force casting like `Type as unknown as NewType`. We want to focus only on `deliveryMethods` there. It might not be okay in application code because we totally replace type, its maybe like any, but I think for this case it will be ok. What can you say about it?

Reply:
refactored